core/spec/models/spree/stock/estimator_spec.rb

```diff
@@ -140,6 +140,32 @@ module Stock
             expect(shipping_rates.first.tax_rate).to eq(tax_rate)
           end
         end
+
+        it 'uses the configured shipping rate selector' do
+          class Spree::Stock::TestSelector; end;
```

Comment:
quick idea: https://gist.github.com/jordan-brough/79a8cbc7d115513d1ad7 (addresses the global mutation also)